dxr/indexers.py

```diff
@@ -326,12 +329,8 @@ def _line_offsets(self):
             if not self.contains_text():
                 raise ValueError("Can't get line offsets for a file that isn't"
                                  " text.")
-            lines = self.contents.splitlines(True) if self.contents else []
-            self._line_offset_list = []
-            chars = 0
-            for i in xrange(0, len(lines)):
-                self._line_offset_list.append(chars)
-                chars += len(lines[i])
+            lines = self.contents.splitlines(True) if self.contents is not None else []
+            self._line_offset_list = list(cumulative_sum(imap(len, lines)))
```

Comment:
And I notice the offset-building happens two more times in test_lines, so definitely should make the function.